ts/Extensions/FullScreen.ts

```diff
@@ -253,12 +254,16 @@ class Fullscreen {
                 function (): void {
                     // Handle lack of async of browser's fullScreenChange event.
                     if (fullscreen.isOpen) {
+                        fireEvent(chart, 'beforeFullscreenClose');
                         fullscreen.isOpen = false;
                         fullscreen.close();
+                        fireEvent(chart, 'afterFullscreenClose');
                     } else {
+                        fireEvent(chart, 'beforeFullscreenOpen');
                         chart.setSize(null, null, false);
                         fullscreen.isOpen = true;
                         fullscreen.setButtonText();
+                        fireEvent(chart, 'afterFullscreenOpen');
```

Comment:
Close events are not working when pressing the exporting menu button (they work only on `escape` button).

I suggest firing events at the beginning and end of the `open` and `close` methods.